src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/OutputFactory.java

```diff
@@ -148,7 +149,23 @@ private String getRuntimeHelperSource() throws URISyntaxException, IOException {
         URL runtimeHelper = OutputFactory.class.getResource("resources/RuntimeHelper.java.template");
         return (pkgName.isEmpty()? "" : "package " + pkgName + ";\n") +
                         String.join("\n", Files.readAllLines(Paths.get(runtimeHelper.toURI())))
-                                .replace("${C_LANG}", C_LANG_CONSTANTS_HOLDER);
+                                .replace("${LOADLIBRARIES}", generateLoadLibraries());
+    }
+
+    private String generateLoadLibraries() {
+        StringBuilder buf = new StringBuilder();
+        buf.append("RuntimeHelper.loadLibraries(new String[] {");
```

Comment:
Instead of calling load libraries, we can just emit calls to System.load/loadLibraries here